File: java/src/tasks/Task20.java
package tasks;

import java.util.HashMap;
import java.util.Stack;

public class Task20 {
    public static
    class Solution {
        public boolean isValid(String s) {
            if ("".equals(s)) {
                return true;
            }

            Stack<Character> stack = new Stack<>();
            HashMap<Character, Character> brackets = new HashMap<>();

            brackets.put(')', '(');
            brackets.put('}', '{');
            brackets.put(']', '[');
            System.out.println("test");
            for (int i = 0; i < s.length(); i++) {
                Character currentChar = s.charAt(i);
                switch (currentChar) {
                    case '(':
                    case '{':
                    case '[':
                        stack.push(s.charAt(i));
                        break;
                    case ')':
                    case '}':
                    case ']':
                        if (stack.size() == 0) {
                            return false;
                        }

                        if (stack.pop() != brackets.get(currentChar)) {
                            return false;
                        }

                        break;
                }
            }

            return stack.size() == 0;
        }
    }
}

File: go/tasks/task_20.go
package tasks

func isValid(s string) bool {
	var stack []byte
	var brackets = map[byte]byte{
		')': '(',
		'}': '{',
		']': '[',
	}

	for i := 0; i < len(s); i++ {
		if s[i] == '(' || s[i] == '{' || s[i] == '[' {
			stack = append(stack, s[i])
			continue
		}

		if s[i] == ')' || s[i] == '}' || s[i] == ']' {
			if len(stack) == 0 {
				return false
			}

			if stack[len(stack)-1] != brackets[s[i]] {
				return false
			}

			stack = stack[:len(stack)-1]
		}

	}

	return len(stack) == 0
}
